### PR TITLE
#21 run component-inspector with react 15.* [WIP]

### PR DESCRIPTION
**Pull request by Mavrin**

### CODE REVIEW

**Comment by lahmatiy** [comment]
Thank you a lot!
How about make it work with 15.* and 0.14.* in the same time?

File: examples/react-todomvc/server.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ var server = new WebpackDevServer(webpack(config), {
   }
 });
 
-server.listen(3000, 'localhost', function (err) {
+server.listen(9000, 'localhost', function (err) {
```
Comment (lahmatiy) [inline examples/react-todomvc/server.js]: Are you sure we need to change this?

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "basisjs-tools": "^1.6.1"
   },
   "scripts": {
-    "build": "basis build --pack",
+    "build": "basis build",
```
Comment (lahmatiy) [inline package.json]: Believe it should be reverted

File: src/api-adapter/react.js [hunk 3]
```diff
@@ -21,54 +20,50 @@ function unwrap(value, getDevInfo) {
...
-function isComponentRootNode(node){
+function isComponentRootNode(node) {
   var element = getReactElementByNode(node);
-
-  if (element &&
-      element._currentElement &&
-      typeof element._currentElement != 'number' &&
-      typeof element._currentElement != 'string' &&
-      typeof element._currentElement.type != 'string') {
+  if (element && element._currentElement && typeof element._currentElement != "number" && typeof element._currentElement != "string") {
```
Comment (lahmatiy) [inline src/api-adapter/react.js]: I see only `typeof element._currentElement.type != 'string'` is removed. May be revert it with no one condition only?

File: src/api-adapter/react.js [hunk 4]
```diff
@@ -21,54 +20,50 @@ function unwrap(value, getDevInfo) {
...
     return true;
   }
-
   return false;
 }
 
+function tryGetName(element, name) {
+  var currentElement = element._currentElement;
+  if (currentElement && currentElement._owner && currentElement._owner._currentElement) {
+      if (currentElement._owner._currentElement.type) {
```
Comment (lahmatiy) [inline src/api-adapter/react.js]: offset

File: src/api-adapter/react.js [hunk 5]
```diff
@@ -94,8 +89,8 @@ var InstanceView = Node.subclass({
 
 function getAdditionalInstanceInfo(element){
   var instanceRootNode = this.getInstanceRootNode(element);
-  var instance = element && element._instance;
-  var cls = instance && instance.constructor;
+  var instance = element && element._currentElement && element._currentElement._owner && element._currentElement._owner._currentElement.type;
```
Comment (lahmatiy) [inline src/api-adapter/react.js]: Looks like we don't need for `instance` variable anymore

**Comment by Mavrin** [comment]
Yes it is possible to make that work with react 15 and 14. I think we should support two latest major version react.
This pull request is not completed .
I need figure out component-inspector api and need write test on adapter methods.

**Comment by Mavrin** [comment]
@lahmatiy Looks like I have done.

**Comment by lahmatiy** [comment]
@Mavrin Cool! Will try tomorrow.

**Comment by lahmatiy** [comment]
I merged PR. But looks to early. Now doesn't work as expected – wrongly resolve element is component root. Compare the screenshots:

React 14
react 14

React 15
react 15

**Comment by Mavrin** [comment]
Ok. I will check out it.

**Pull request merged**